src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -22,11 +24,27 @@ public class Workspace
     {
         #region Private Fields
 
-        private static readonly string[] s_psFilePatterns = new []
+        private static readonly string[] s_psFileExtensions = new []
         {
-            "*.ps1",
-            "*.psm1",
-            "*.psd1"
+            ".ps1",
+            ".psm1",
+            ".psd1"
+        };
+        // .Net Core doesn't appear to use the same three letter pattern matching rule although the docs
+        // suggest it should be find the '.ps1xml' files because we search for the pattern '*.ps1'
+        // ref https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=netcore-2.1#System_IO_Directory_GetFiles_System_String_System_String_System_IO_EnumerationOptions_
+        private static readonly string[] s_psFileExtensionsDotNetFramework = new []
+        {
+            ".ps1",
+            ".psm1",
+            ".psd1",
+            ".ps1xml"
```

Comment:
The parser etc. _shouldn't_ understand ps1xml.  It's XML not PowerShell language.  However I kept the ps1xml files in as this was the behaviour of the `EnumeratePSFiles` method before I added the filtering. If I removed them then I would be modifying existing known behaviour as per the characterisation tests.

There's no reason this detection code couldn't be removed later if needs be

Reply:
That's fair. What happens when you just use a single list that has all 4 file types for both Desktop and Core?

Does Core behave differently with the added file extension?

Note: what Im suggesting should hopefully have no functionality change, but the code would be cleaner since we wouldn't have to check the version of .NET

Reply:
> What happens when you just use a single list that has all 4 file types for both Desktop and Core?

> ... should hopefully have no ...

From a contributor point-of-view, unfortunately I can't make those determinations yet.  There appear to be no tests for this scenario (unit or integration) so I can't guarantee anything.

Do you want me to go hunting and add these characterisation tests first?

Reply:
Ok...went hunting, the current tests are mocking the responses so I'd need to write new tests for this 😿

Reply:
I think I'm confused... but...

You added a comment that says:
```
// .Net Core doesn't appear to use the same three letter pattern matching rule although the docs
// suggest it should be find the '.ps1xml' files because we search for the pattern '*.ps1'.
```
were you able to prove this?

Reply:
@TylerLeonhardt 100% prove it.  That's what the characterisation tests in the first commit are for.  Those tests will pass on master right now and shows there is a difference.

I also test for `.pscc`, `.psrc`, `.cdxml` and `.txt` just in case.

Reply:
> the current tests are mocking the responses

I'll explain this in more detail

The characterisation tests I wrote for EnumeratePSFiles, use actual files on disk (test/PowerShellEditorServices.Test/Fixtures/Workspace). Strictly speaking they're integration tests to prove for a given file system layout it expects X files to be returned.

However the tests for [Definitions](https://github.com/PowerShell/PowerShellEditorServices/blob/bfb4e8e5008a7d48397d1d0cfccd98348064cd78/test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs#L140-L212) and [References](https://github.com/PowerShell/PowerShellEditorServices/blob/bfb4e8e5008a7d48397d1d0cfccd98348064cd78/test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs#L238-L280) use the shared test setup mocks in PowerShellEditorServices/test/PowerShellEditorServices.Test.Shared and never actually call EnumeratePSFiles i.e. the mocking hides the actual behaviour of EnumeratePSFiles

As far as I can tell there are no tests that exercise the integration (and I use that word on purpose) between the three call sites that I found and EmueratePSFiles

1.
`LanguageSevice.cs -> HandleDefinitionRequestAsync` calls
`LanguageSevice.cs -> GetDefinitionOfSymbolAsync` calls
`EnumeratePSFiles`

2.
`LanguageSevice.cs -> HandleReferencesRequestAsync` calls
`LanguageSevice.cs -> FindReferencesOfSymbolAsync` calls
`EnumeratePSFiles`

3. ReferencesCodeLensProvider is use in the CodeLensFeature, PesterCodeLensProvider and ReferencesCodeLensProvider
`ReferencesCodeLensProvider.cs -> ResolveCodeLensAsync` calls
`LanguageSevice.cs -> FindReferencesOfSymbolAsync` calls
`EnumeratePSFiles`

Reply:
> 100% prove it. That's what the characterization tests in the first commit are for. Those tests will pass on master right now and shows there is a difference.

So if you add .ps1xml to Core's list of extensions... we could prove that it yields the same result as without it, right?

Reply:
I think you're missing which tests are proving what.

I'll try explaining myself differently.  All of what I'm writing about below relates the the head of master, NOT my PR i.e. I will be explaining the current behaviour of PSES (Apologies if you already know this stuff):

---

The scenario is:
- "User in VSCode requests the definition of a symbol"
- VSCode sends a 'textDocument/definition' to PSES
- PSES receives the 'textDocument/definition' message and calls the 'HandleDefinitionRequestAsync' handler
...

Inside the 'HandleDefinitionRequestAsync' method
- The handler parses the document to find the name of symbol the user requested
- The handler then calls 'GetDefinitionOfSymbolAsync' to "please find all definitions of symbol XXXXXXX in the workspace"
...

Inside the 'GetDefinitionOfSymbolAsync' method
- First we search any "Referenced Files" for definitions of the symbol XXXXXXX
- If we find a definition then return that, otherwise search the entire workspace
- We call EnumeratePSFiles in the workspace and then for each file we parse and generate an AST.
  - For each file's AST when search for definitions of the symbol XXXXXXX

Inside the 'EnumeratePSFiles' method
- We search the entire workspace, swallowing most errors, searching for PowerShell "related" files and return the resolved file list

---

That's the workflow for that one request.

The characterisation tests I wrote in the PR prove that: Given the same directory structure and files, the output from 'EnumeratePSFiles' method changes between the two .Net distributions.  This is a fault (IMO) of the .Net library, not PSES.

The current test suite does actually call 'GetDefinitionOfSymbolAsync' HOWEVER all of the references and definitions are pre-loaded as test fixtures and there are no tests using on disk file structures.  That is, the odd behaviour of EnumeratePSFiles is invisible because it's never called in the tests.

---

Now it is currently my _feeling_ that ps1xml files are effectively ignored because in 'GetDefinitionSymbolAsync', any parse errors are silently ignored (https://github.com/PowerShell/PowerShellEditorServices/blob/master/src/PowerShellEditorServices/Language/LanguageService.cs#L463-L467)

BUT there are NO tests to prove that this is case. Also, it's calling Parser.ParseFile(.... which:
1. Is a class in the PowerShell SDK, not in the PSES project.  Unknown what the testing coverage is there and what contracts it provides us
2. It's passing in a filename, not file content.  Therefore any tests here need to use on-disk files, not mocks.

---

TBH at this point I'd need to write characterisation tests for;
A) The 'GetDefinitionOfSymbolAsync' method and throw different file structures at it.

OR

B) The Parser.ParseFile(...) method and throw different files structures at it and prove that ps1xml files generate errors and no parser tokens.

That way we can confirm the behaviour of what happens with ps1xml files and therefore make a decision on whether to keep the ps1xml or get rid of it.

---

Also none of this should stop the PR. This PR preserves the existing (while odd) behaviour of EnumeratePSFiles with a new implementation, and adds new behaviours (File ignores, reparse point ignores)

Reply:
Ok I think I understand now. 

I had thought that you added this in so that Core and Framework could be _consistent_ but instead you are adding it in to keep the current behavior which is that Core and Framework yield different results.

Reply:
Correct